src/realm/object-store/shared_realm.hpp

```diff
@@ -350,6 +350,11 @@ class Realm : public std::enable_shared_from_this<Realm> {
     // commit has not yet reached permanent storage.
     bool is_in_async_transaction() const noexcept;
 
+    void set_async_error_handler(const std::function<void(AsyncHandle, std::exception_ptr)>& hndlr)
```

Comment:
I think you could take a `UniqueFunction` by value or by `&&` here. std::move-ing a const ref to a std::function feels like some real mixed messages here.

Reply:
Yes - I am aware of that. I will replace all std::function with UniqueFunction in a subsequent PR